service/storage_service.cc

```diff
@@ -998,6 +998,20 @@ class topology_coordinator {
         }
     }
 
+    future<std::pair<node_to_work_on, bool>> global_token_metadata_barrier(node_to_work_on&& node) {
+        bool exec_command_res;
+        std::tie(node, exec_command_res) = co_await exec_global_command(std::move(node),
```

Comment:
`exec_global_command()` only works on normal nodes, but we want the barrier to work with all members.

Reply:
Why? None normal nodes do not suppose to participate in queries.

Reply:
Bootstrapping node participates as a replica. So we want fence to take effect on them.

Reply:
But not as a coordinator and we drain coordinator requests here, not replicas.

Reply:
Ok, but what about the next invocation, which sends the fence command?

Reply:
The fence is needed to fail commands that we did not managed to drain. But non of them were for the bootstrapping node. First barrier drains writes that goes to old nodes only, second drains reads that goes throw the old nodes.

Reply:
Ok, it seems to work given current algorithm. It may break if we add other kinds of modes which rely on the promise (which we don't provide) that after the barrier only requests using latest topology version will be admitted to any node.